### PR TITLE
use a private endpointDiscoveryResponse to generate eds response

### PR DESCRIPTION
**Pull request by hzxuzhonghu**

### CODE REVIEW

File: pilot/pkg/proxy/envoy/v2/eds.go [hunk 1]
```diff
@@ -770,12 +746,16 @@ func (s *DiscoveryServer) removeEdsCon(clusterName string, node string, connecti
 	}
 }
 
-// FetchEndpoints implements xdsapi.EndpointDiscoveryServiceServer.FetchEndpoints().
-func (s *DiscoveryServer) FetchEndpoints(ctx context.Context, req *xdsapi.DiscoveryRequest) (*xdsapi.DiscoveryResponse, error) {
-	return nil, errors.New("not implemented")
-}
+func endpointDiscoveryResponse(loadAssignments []*xdsapi.ClusterLoadAssignment) *xdsapi.DiscoveryResponse {
+	out := &xdsapi.DiscoveryResponse{
+		TypeUrl:     EndpointType,
```
Comment (ymesika) [inline pilot/pkg/proxy/envoy/v2/eds.go]: Can you also keep the comments from `endpoints`?

Comment (hzxuzhonghu) [reply pilot/pkg/proxy/envoy/v2/eds.go]: > // All resources for EDS ought to be of the type ClusterLoadAssignment	

This is useless

Comment (ymesika) [reply pilot/pkg/proxy/envoy/v2/eds.go]: The other one

Comment (hzxuzhonghu) [reply pilot/pkg/proxy/envoy/v2/eds.go]: done

**Pull request merged**